src/Simplify.cpp

```diff
@@ -308,12 +309,27 @@ class Simplify : public IRMutator {
         if (const int64_t *i = as_const_int(e)) {
             *min_val = *max_val = *i;
             return true;
+        } else if (const uint64_t *u = as_const_uint(e)) {
+            if (t.bits() < 64) {
```

Comment:
It's probably better to use Int(64).can_represent(*u). It also does the overflow check for you.